### PR TITLE
fix(Provider): Honor disableAnimations

### LINKED ISSUE #2086: Unable to disable animations
### Steps
```jsx
<Provider disableAnimations>
  <AnimatedComponent>
</Provider>
```

### Expected Result
`AnimatedComponent` is NOT animated

### Actual Result
It still animates 💣

### Version
0.40.2

### Testcase
https://codesandbox.io/s/xbkuv?module=/example.js

### PR DESCRIPTION
**Pull request by miroslavstastny**
Fixes #2086

### CODE REVIEW

File: packages/react/src/lib/felaDisableAnimationsPlugin.ts [hunk 1]
```diff
@@ -24,16 +24,17 @@ export default () => {
       return Object.keys(styles).reduce((acc, cssPropertyName) => {
         const cssPropertyValue = styles[cssPropertyName]
 
+        if (animationProps.indexOf(cssPropertyName) !== -1) {
```
Comment (miroslavstastny) [inline packages/react/src/lib/felaDisableAnimationsPlugin.ts:27]: We want to remove `animationProps` even if they are objects.

File: packages/react/src/lib/renderComponent.tsx [hunk 2]
```diff
@@ -231,6 +231,7 @@ const renderComponent = <P extends {}>(
   const direction = rtl ? 'rtl' : 'ltr'
   const felaParam = {
     theme: { direction },
+    disableAnimations,
```
Comment (mnajdova) [inline packages/react/src/lib/renderComponent.tsx:242]: We have check in the plugin for this:
```
if (props && props.disableAnimations && type === 'RULE')
```

Do we really need this one here too?

Comment (miroslavstastny) [reply packages/react/src/lib/renderComponent.tsx:242]: But the `felaParam` here is the `props` in the plugin. That was the cause of the issue.
We originally passed or params to fela but that changed in #1957 and it caused this bug.

Comment (mnajdova) [reply packages/react/src/lib/renderComponent.tsx:242]: Got it! Thanks for the clarification

**Review by mnajdova — APPROVED** [review]
Please add changelog entry before merging

**Pull request merged**